modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -2249,144 +2324,179 @@ private Expression<?>[] _getSelectExpressions(
-					dynamicObjectRelationshipMappingTable,
-					primaryKeyColumn2.eq(
-						relatedDynamicObjectDefinitionTable.
-							getPrimaryKeyColumn()));
-
-				Column<DynamicObjectRelationshipMappingTable, Long>
-					primaryKeyColumn1 =
-						dynamicObjectRelationshipMappingTable.
-							getPrimaryKeyColumn1();
-
-				predicate = primaryKeyColumn1.eq(
-					dynamicObjectDefinitionTable.getPrimaryKeyColumn());
-			}
-
-			List<String> oDataFilterStrings = TransformUtil.transform(
-				(List<ObjectFilter>)objectFieldSettingsValues.get("filters"),
-				objectFilter -> {
-					if (StringUtil.equals(
-							objectFilter.getFilterType(),
-							ObjectFilterConstants.TYPE_CURRENT_USER)) {
-
-						objectFilter.setJSON(
-							JSONUtil.put(
-								"currentUserId",
-								PrincipalThreadLocal.getUserId()
-							).toString());
-					}
+				List<String> oDataFilterStrings = TransformUtil.transform(
+					(List<ObjectFilter>)objectFieldSettingsValues.get(
+						"filters"),
+					objectFilter -> {
+						if (StringUtil.equals(
+								objectFilter.getFilterType(),
+								ObjectFilterConstants.TYPE_CURRENT_USER)) {
+
+							objectFilter.setJSON(
+								JSONUtil.put(
+									"currentUserId",
+									PrincipalThreadLocal.getUserId()
+								).toString());
+						}
+
+						ObjectFilterParser objectFilterParser =
+							_objectFilterParserServiceRegistry.
+								getObjectFilterParser(
+									objectFilter.getFilterType());
 
-					ObjectFilterParser objectFilterParser =
-						_objectFilterParserServiceRegistry.
-							getObjectFilterParser(objectFilter.getFilterType());
+						return objectFilterParser.parse(objectFilter);
+					});
 
-					return objectFilterParser.parse(objectFilter);
-				});
+				for (String oDataFilterString : oDataFilterStrings) {
+					predicate = predicate.and(
+						_filterPredicateFactory.create(
+							oDataFilterString,
+							relatedObjectDefinition.getObjectDefinitionId()));
+				}
 
-			for (String oDataFilterString : oDataFilterStrings) {
-				predicate = predicate.and(
-					_filterPredicateFactory.create(
-						oDataFilterString,
-						relatedObjectDefinition.getObjectDefinitionId()));
+				selectExpressions.add(
+					DSLQueryFactoryUtil.scalarSubDSLQuery(
+						joinStep.where(predicate),
+						DynamicObjectDefinitionTable.getJavaClass(
+							objectField.getDBType()),
+						objectField.getName(),
+						DynamicObjectDefinitionTable.getSQLType(
+							objectField.getDBType())));
 			}
+			else if (objectField.compareBusinessType(
+						ObjectFieldConstants.BUSINESS_TYPE_FORMULA)) {
 
-			selectExpressions.add(
-				DSLQueryFactoryUtil.scalarSubDSLQuery(
-					joinStep.where(predicate),
-					DynamicObjectDefinitionTable.getJavaClass(
-						objectField.getDBType()),
-					objectField.getName(),
-					DynamicObjectDefinitionTable.getSQLType(
-						objectField.getDBType())));
+				Object script = objectFieldSettingsValues.get("script");
+
+				if (script == null) {
+					break;
```

Comment:
@guilhermedcamacho just to confirm with you that this logic is correct, i.e. we should use `break` instead of `continue`..

Reply:
It's not a big problem, but it should be `continue`, otherwise there is a small possibility that the object has another formula field without the expression filled, which could cause a problem in this logic.
I will wait for Marcela's review to make this change, thanks!

Reply:
Hey @carolmariaabb , it's not a big problem, but it should be `continue`, otherwise there is a small possibility that the object has another formula field without the expression filled, which could cause a problem in this logic.
I was changing between `continue` and `break` to try to simulate this problem, but I forgot to leave as `continue`. :disappointed: 
I will wait for @marcelabc 's review to make this change, thanks!

Reply:
Thanks for explaining, @guilhermedcamacho 😄